logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/call/EndCallUseCaseTest.kt

```diff
@@ -55,5 +64,65 @@ class EndCallUseCaseTest {
             .function(callRepository::updateIsCameraOnById)
             .with(eq(conversationId.toString()), eq(false))
             .wasInvoked(once)
+
+        verify(callRepository)
+            .function(callRepository::persistMissedCall)
+            .with(eq(conversationId.toString()))
+            .wasNotInvoked()
+    }
+
+    @Test
+    fun givenNonEstablishedCall_whenEndCallIsInvoked_thenSaveMissedCall() = runTest {
+        given(callRepository)
+            .suspendFunction(callRepository::persistMissedCall)
+            .whenInvokedWith(eq(conversationId))
+            .thenDoNothing()
+
+        given(callRepository)
+            .suspendFunction(callRepository::establishedCallsFlow)
+            .whenInvoked().then { flowOf(listOf(call.copy(status = CallStatus.INCOMING))) }
```

Comment:
`establishedCallsFlow` returns only calls with `ESTABLISHED` or `ANSWERED` status, for incoming calls we should use `::incomingCallsFlow`

Reply:
Yeah right . I will update it to return empty list